### ISSUE TITLE
Possible problems with finidat file clmi.F2000.2000-01-01.ne120pg3_mt13_simyr2000_c200728.nc

### ISSUE THREAD

**Issue by billsacks** [ESCOMP/CTSM#1931]
### Brief summary of bug

@mvertens reports that, when running a case that interpolates from clmi.F2000.2000-01-01.ne120pg3_mt13_simyr2000_c200728.nc, the model dies with a "CheckSnowConsistency ERROR" message.

### General bug information

**CTSM version you are using:** @mvertens is using a version similar to ctsm5.1.dev114

**Does this bug cause significantly incorrect results in the model's science?** No

**Configurations affected:** Cases using finidat = clmi.F2000.2000-01-01.ne120pg3_mt13_simyr2000_c200728.nc

### Details of bug

I introduced this check a few years ago when I was reworking how snow state variables are handled. I did get one report from @olyson  (passing something on from a user) a little over a year ago of this being triggered, but he didn't provide any details about the initial conditions file or the details of the error message. What I said to him at the time was:

> As for this particular error: I checked the tag where this check was introduced (ctsm1.0.dev046) and I didn't need to change any of the out-of-the-box initial conditions files. And I don't think I have heard of anyone running into this problem before. I can see how this problem would arise if there were problems in the initial conditions file – specifically, non-zero h2osoi_ice or h2osoi_liq in non-existent snow layers – though I'm not sure how that would have come to be in the first place... maybe some bug in the history of the code that produced that initial conditions file? I could be missing some possible cause, though.

I suspect that I haven't heard this reported more because this is only checked in debug mode, and it's probably rare to run in debug mode with this particular initial conditions file. I think there are a few possible paths forward:

(1) Interpolate from a different initial conditions file

(2) Run for a bit either in non-debug mode (I think this condition is only checked in debug mode, because it is a relatively expensive check to be running all the time) or with the given error removed... it's possible that this will work itself out after running for a bit. It's quite possible that, if all you have are roundoff-level issues like this, then there won't be any real problems, so it might even be safe to remove this check for the sake of what you're doing... but I can't remember well enough how important this check is for the correct operation of the code.

(3) Do some pre-processing of this initial conditions file so that very small snow values are changed to zero... but this might be a pain to get right

### Important details of your setup / configuration so we can reproduce the bug

@mvertens says:

> I am using a model resolution of f05_f05_mg17 (this is a CMCC resolution) and effectively
ctsm5.1.dev114 (with a few minor changes needed for CMCC).
It is trying to use the following finidat file to map to this resolution:
/data/inputs/CESM/inputdata/lnd/clm2/initdata_map/[clmi.F2000.2000-01-01.ne120pg3_mt13_simyr2000_c200728.nc](http://clmi.f2000.2000-01-01.ne120pg3_mt13_simyr2000_c200728.nc/)

### Important output or errors that show the problem

@mvertens reports:

> I am getting the following error:

```
 CheckSnowConsistency ERROR: col has non-zero h2osoi_ice or h2osoi_liq outside resolved snow layers
 (Called from: lnd2atm_minimal: bulk)
 c, j, snl, h2osoi_ice, h2osoi_liq =            1         -11         -10
  3.785766995733679E-270  3.785766995733679E-270
iam = 0: local  column   index = 1
iam = 0: global column   index = 1
iam = 0: global landunit index = 1
iam = 0: global gridcell index = 1
iam = 0: gridcell longitude    =    0.0000000
iam = 0: gridcell latitude     =  -90.0000000
iam = 0: column   type         = 1
iam = 0: landunit type         = 1
 ENDRUN:
 ERROR:
 CheckSnowConsistency ERROR: col has non-zero h2osoi_ice or h2osoi_liq outside 
resolved snow layers
```

### Suggested path forward

**We should verify that this error appears in an out-of-the-box case with the latest version of CTSM. If it does, then I'd probably suggest just pointing to a different initial conditions file in place of this one.**

**Comment by billsacks** [comment]
I tried to reproduce this from latest CTSM master with SMS_D_Ld1.ne120pg3_t13.I2000Clm50Sp.cheyenne_intel, where I set `./xmlchange LND_TUNING_MODE=clm5_0_cam6.0`; this used the finidat file that caused problems for @mvertens . However, this run completed without a problem. It's possible that the different outcome is due to the different resolution, or maybe due to the more recent tag, or maybe due to machine differences. In any case, I'm going to close this as a wontfix until we get a reproducible bug report for out-of-the-box CTSM.
